extensions/data-transfer/portability-data-transfer-google/src/test/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosImporterTest.java

```diff
@@ -329,44 +329,121 @@ public void importPhotoInTempStore() throws Exception {
     Mockito.doNothing().when(jobStore).removeData(any(), anyString());
 
     GooglePhotosImporter googlePhotosImporter =
-        new GooglePhotosImporter(
-            null, jobStore, null, null, googlePhotosInterface, null, null, 1.0);
+            new GooglePhotosImporter(
+                    null, jobStore, null, null, googlePhotosInterface, null, null, 1.0);
 
     BatchMediaItemResponse batchMediaItemResponse =
-        new BatchMediaItemResponse(
-            new NewMediaItemResult[] {buildMediaItemResult("token1", Code.OK_VALUE)});
+            new BatchMediaItemResponse(
+                    new NewMediaItemResult[] {buildMediaItemResult("token1", Code.OK_VALUE)});
 
     Mockito.when(googlePhotosInterface.createPhotos(any(NewMediaItemUpload.class)))
-        .thenReturn(batchMediaItemResponse);
+            .thenReturn(batchMediaItemResponse);
 
     UUID jobId = UUID.randomUUID();
 
     googlePhotosImporter.importPhotoBatch(
-        jobId,
-        Mockito.mock(TokensAndUrlAuthData.class),
-        Lists.newArrayList(photoModel),
-        executor,
-        NEW_ALBUM_ID);
+            jobId,
+            Mockito.mock(TokensAndUrlAuthData.class),
+            Lists.newArrayList(photoModel),
+            executor,
+            NEW_ALBUM_ID);
     Mockito.verify(jobStore, Mockito.times(0)).removeData(any(), anyString());
     Mockito.verify(jobStore, Mockito.times(1)).getStream(any(), anyString());
   }
+
+  @Test
+  public void importPhotoFailedToFindAlbum() throws Exception {
+    PhotoModel photoModel =
+            new PhotoModel(
+                    PHOTO_TITLE,
+                    IMG_URI,
+                    PHOTO_DESCRIPTION,
+                    JPEG_MEDIA_TYPE,
+                    "oldPhotoID1",
+                    OLD_ALBUM_ID,
+                    true);
+
+    Mockito.when(googlePhotosInterface.uploadPhotoContent(any())).thenReturn("token1", "token2");
+    JobStore jobStore = Mockito.mock(LocalJobStore.class);
+    Mockito.when(jobStore.getStream(any(), any()))
+            .thenReturn(
+                    new TemporaryPerJobDataStore.InputStreamWrapper(
+                            new ByteArrayInputStream("TestingBytes".getBytes())));
+    googlePhotosImporter =
+            new GooglePhotosImporter(
+                    null, jobStore, null, null, googlePhotosInterface, imageStreamProvider, monitor, 1.0);
+    Mockito.when(googlePhotosInterface.createPhotos(any(NewMediaItemUpload.class)))
+            .thenThrow(new IOException("The provided ID does not match any albums"));
+
+    GoogleAlbum responseAlbum = new GoogleAlbum();
+    Mockito.when(googlePhotosInterface.getAlbum(any()))
+            .thenReturn(responseAlbum);
+
+    long bytes = googlePhotosImporter.importPhotoBatch(
+            uuid,
+            Mockito.mock(TokensAndUrlAuthData.class),
+            Lists.newArrayList(photoModel),
+            executor,
+            NEW_ALBUM_ID);
+
+    // didn't throw
+    assertEquals(0, bytes);
```

Comment:
probably good to test zero bytes as you are, but also maybe encode your comment "didn't throw" as a `assertDoesNotThrow`? eg:

```java
long bytes;

assertDoesNotThrow(() -> {
  bytes = googlePhotosImporter.importPhotoBatch(...);
});

assertEquals(0, bytes);
```